internal/services/recoveryservices/backup_protected_vm_resource.go

```diff
@@ -106,17 +110,52 @@ func resourceRecoveryServicesBackupProtectedVMCreateUpdate(d *pluginsdk.Resource
 		},
 	}
 
-	if _, err = client.CreateOrUpdate(ctx, vaultName, resourceGroup, "Azure", containerName, protectedItemName, item); err != nil {
-		return fmt.Errorf("creating/updating Azure Backup Protected VM %q (Resource Group %q): %+v", protectedItemName, resourceGroup, err)
-	}
+	requireAdditionalUpdate := d.Get("protection_stopped").(bool)
+	skipNormalUpdate := d.Get("protection_stopped").(bool) && !d.IsNewResource()
 
-	resp, err := resourceRecoveryServicesBackupProtectedVMWaitForStateCreateUpdate(ctx, client, vaultName, resourceGroup, containerName, protectedItemName, d)
-	if err != nil {
-		return err
+	if !skipNormalUpdate {
+		if _, err = client.CreateOrUpdate(ctx, vaultName, resourceGroup, "Azure", containerName, protectedItemName, item); err != nil {
+			return fmt.Errorf("creating/updating Azure Backup Protected VM %q (Resource Group %q): %+v", protectedItemName, resourceGroup, err)
+		}
+
+		resp, err := resourceRecoveryServicesBackupProtectedVMWaitForStateCreateUpdate(ctx, client, vaultName, resourceGroup, containerName, protectedItemName, d)
+		if err != nil {
+			return err
+		}
+
+		id := strings.Replace(*resp.ID, "Subscriptions", "subscriptions", 1) // This code is a workaround for this bug https://github.com/Azure/azure-sdk-for-go/issues/2824
+		d.SetId(id)
 	}
 
-	id := strings.Replace(*resp.ID, "Subscriptions", "subscriptions", 1) // This code is a workaround for this bug https://github.com/Azure/azure-sdk-for-go/issues/2824
-	d.SetId(id)
+	if requireAdditionalUpdate {
+		updateInput := backup.ProtectedItemResource{
+			Properties: &backup.AzureIaaSComputeVMProtectedItem{
+				ProtectionState:  backup.ProtectionStateProtectionStopped,
+				SourceResourceID: utils.String(vmId),
+			},
+		}
+
+		resp, err := client.CreateOrUpdate(ctx, vaultName, resourceGroup, "Azure", containerName, protectedItemName, updateInput)
+		if err != nil {
+			return fmt.Errorf("creating/updating Azure Backup Protected VM %q (Resource Group %q): %+v", protectedItemName, resourceGroup, err)
+		}
+
+		locationURL, err := resp.Response.Location()
+		if err != nil || locationURL == nil {
+			return fmt.Errorf("creating/updating Azure Backup Protected VM %q (Resource Group %q): Location header missing or empty", protectedItemName, resourceGroup)
+		}
+
+		parsedLocation, err := azure.ParseAzureResourceID(handleAzureSdkForGoBug2824(locationURL.Path))
+		if err != nil {
+			return err
+		}
+
+		opState := resourceRecoveryServicesBackupProtectedVMOperationRefreshFunc(ctx, meta.(*clients.Client).RecoveryServices.BackupOperationResultsClient, d.Timeout(pluginsdk.TimeoutCreate), vaultName, resourceGroup, parsedLocation.Path["operationResults"])
+		if _, err := opState.WaitForStateContext(ctx); err != nil {
+			return fmt.Errorf("creating/updating Azure Backup Protected VM %q (Resource Group %q): %+v", protectedItemName, resourceGroup, err)
+		}
```

Comment:
is there a Swagger bug to highlight this is an LRO? `hashicorp/go-azure-sdk` has helpers to do this for us: https://github.com/hashicorp/go-azure-sdk/blob/main/sdk/client/resourcemanager/poller.go#L12

Reply:
just opened one,https://github.com/Azure/azure-rest-api-specs/issues/22758
The poller is for go-azure-sdk may not be suitable for this?